test/new-e2e/tests/agent-shared-components/forwarder/nss_failover_test.go

```diff
@@ -0,0 +1,302 @@
+			return nil, err
+		}
+
+		fiExporter2, err := aws.NewEcsFakeintake(awsEnv, fakeintakeparams.WithName(fakeintake2Name), fakeintakeparams.WithoutLoadBalancer())
+		if err != nil {
+			return nil, err
+		}
+
+		agentInstaller, err := agent.NewInstaller(vm, agentOptions...)
+		if err != nil {
+			return nil, err
+		}
+
+		return &multiFakeIntakeEnv{
+			VM:          client.NewPulumiStackVM(vm),
+			Agent:       client.NewPulumiStackAgent(agentInstaller),
+			Fakeintake1: client.NewFakeintake(fiExporter1),
+			Fakeintake2: client.NewFakeintake(fiExporter2),
+		}, nil
+	})
+}
+
+type multiFakeIntakeSuite struct {
+	e2e.Suite[multiFakeIntakeEnv]
+}
+
+func TestMultiFakeintakeSuite(t *testing.T) {
+	e2e.Run(t, &multiFakeIntakeSuite{}, multiFakeintakeStackDef())
+}
+
+// SetupSuite waits for both fakeintakes to be ready before running tests.
+func (v *multiFakeIntakeSuite) SetupSuite() {
+	v.Env() // update the environment outside EventuallyWithT
+
+	// Wait for the fakeintake to be ready to avoid 503
+	require.EventuallyWithT(v.T(), func(c *assert.CollectT) {
+		assert.NoError(c, v.Env().Fakeintake1.Client.GetServerHealth())
+		assert.NoError(c, v.Env().Fakeintake2.Client.GetServerHealth())
+	}, intakeMaxWaitTime, intakeTick)
+}
+
+// BeforeTest flushes both fakeintakes before starting each test.
+func (v *multiFakeIntakeSuite) SetupTest() {
+	v.NoError(v.Env().Fakeintake1.FlushServerAndResetAggregators())
+	v.NoError(v.Env().Fakeintake2.FlushServerAndResetAggregators())
+}
+
+// TestNSSFailover tests that the agent correctly picks-up an NSS change of the intake.
+//
+// The test uses two fakeintakes to represent two backends, and the /etc/hosts file for the NSS source,
+// setting-up an entry for the intake, pointing to the first fakeintake, then changing that entry
+// to point to the second fakeintake without restarting the agent.
+//
+// The test checks that metrics, logs, and flares are properly received by the first intake before
+// the failover, and by the second one after.
+//
+// Note: although the man page of `nsswitch.conf` states that each process using it should only read
+// it once (ie. no reload), glibc and Go reload it periodically
+// cf. https://go-review.googlesource.com/c/go/+/448075
+//
+// TODO: handle APM traces
+func (v *multiFakeIntakeSuite) TestNSSFailover() {
+	// ensure host uses files for NSS
+	enforceNSSwitchFiles(v.T(), v.Env().VM)
+
+	// setup NSS entry for intake
+	fakeintake1IP, err := hostIPFromURL(v.Env().Fakeintake1.URL())
+	v.NoError(err)
+	setHostEntry(v.T(), v.Env().VM, intakeName, fakeintake1IP)
+
+	// configure agent to use the custom intake, set connection_reset_interval, use logs, and processes
+	agentOptions := []agentparams.Option{
+		agentparams.WithAgentConfig(getAgentConfig(connectionResetInterval)),
+		agentparams.WithLogs(),
+		agentparams.WithIntakeHostname(intakeName),
+		agentparams.WithIntegration("custom_logs.d", customLogsConfig),
+	}
+	v.UpdateEnv(multiFakeintakeStackDef(agentOptions...))
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution. -->
**❓ question**
Why don't you do it at setup time ?

Reply:
I mean why not while you define the default run function in `multiFakeintakeStackDef`

Reply:
I need the fakeintake ip address so that I can setup the entry in `/etc/hosts`, if I start the agent before setting up the entry then it will fail to contact the backend.
This is why I first start it without any specific config, do all the setup I need on the VM, and then change the config and restart it